MoviesMVVM/MoviesMVVM/DetailModule/View/FilmViewController.swift

```diff
@@ -154,31 +155,62 @@ final class FilmViewController: UIViewController {
         return button
     }()
 
+    // MARK: - Public property
+
+    weak var coordinator: MainCoordinator?
+    var filmIndex: Int?
+
     // MARK: - Private property
 
-    private var filmInfo: Film?
+    private var filmViewModel: FilmViewModelProtocol?
 
-    // MARK: - Public property
+    // MARK: - Initializer
 
-    var filmIndex: Int?
+    init(filmViewModel: FilmViewModelProtocol) {
+        self.filmViewModel = filmViewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
 
     // MARK: - LifeCycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
-        loadFilmData()
+        filmViewModel?.loadFilmData()
+        updateTableView()
+        alertView()
+        makeImageView()
     }
 
     // MARK: - Private methods
 
-    private func loadFilmData() {
-        guard let index = filmIndex else { return }
-        Service.shared.loadFilm(index: index) { [weak self] result in
-            self?.filmInfo = result
+    private func updateTableView() {
+        filmViewModel?.updateViewData = {
+            guard let film = self.filmViewModel?.filmInfo else { return }
+            DispatchQueue.main.async {
+                self.setupData(data: film)
+            }
+        }
+    }
+
+    private func alertView() {
```

Comment:
```suggestion
    private func bind() {
```
собери все кложуры в 1 м таком методе